### ISSUE TITLE
BUG - Jellyfin - While processing series: string indices must be integers, not 'str'

### ISSUE THREAD

**Issue by JPsychotic** [CollinHeist/TitleCardMaker#390]
### Installation

GitHub - develop branch

### Describe the Bug

When processing one show TCM has an error and it doesn't even download the images of the show :( Metadata of the episodes seems okay. They all have season and episode index.

### Steps to reproduce the behavior

1. 
2.
...

### Preference File

```yaml
options:
  source: ./source/
  series:
  - ./config/sync_series.yml
  card_type: standard
  card_extension: png
  card_dimensions: 1920x1080
  validate_fonts: true
  sync_specials: false
  episode_data_source: jellyfin
  image_source_priority: jellyfin
  language_codes:
  - de
  - en
jellyfin:
  url: https://jellyfin.local/
  api_key: api_key
  username: Jeremy
  verify_ssl: false
  watched_style: unique
  unwatched_style: unique
  sync:
  - sync_mode: append
  - card_directory: ./config/cards
  - file: ./config/sync_series.yml
  libraries:
  - Serien
```

### Series YAML

```yaml
libraries:
  Serien:
    media_server: jellyfin
    path: .\config\cards\tvshows
series:
  Jujutsu Kaisen (2020): {library: Serien, media_directory: .\config\cards\tvshows\Jujutsu Kaisen, jellyfin_id: 15ecda55a0eb2b1bcb1e9d1b4b29a4ac, card_type: anime, extras: {kanji: 呪術廻戦, require_kanji: true}}
```

### Debug Log

[INFO] [08-25-23 07:21:42.499] Reading series YAML file "D:\SourceCode\TitleCardMaker\config\sync_series.yml"..
[INFO] [08-25-23 07:21:42.749] Wrote missing assets to "D:\SourceCode\TitleCardMaker\missing.yml"
[DEBUG] [08-25-23 07:25:11.819] Runtime arguments  :
[DEBUG] [08-25-23 07:25:11.819]        preferences : D:\SourceCode\TitleCardMaker\preferences.yml
[DEBUG] [08-25-23 07:25:11.819]                run : True
[DEBUG] [08-25-23 07:25:11.819]               sync : False
[DEBUG] [08-25-23 07:25:11.819]          frequency : {'interval': 12, 'unit': 'hours'}
[DEBUG] [08-25-23 07:25:11.819]            missing : D:\SourceCode\TitleCardMaker\missing.yml
[DEBUG] [08-25-23 07:25:11.819]                log : INFO
[DEBUG] [08-25-23 07:25:11.819]           no_color : False
[DEBUG] [08-25-23 07:25:11.819] tautulli_frequency : {'interval': 4, 'unit': 'minutes'}
[INFO] [08-25-23 07:25:11.824] Read preference file "D:\SourceCode\TitleCardMaker\preferences.yml"
[DEBUG] [08-25-23 07:25:11.866] Using "magick " ImageMagick command prefix
[DEBUG] [08-25-23 07:25:11.875] Deleted show record database, was version v1.14.2-develop37
[INFO] [08-25-23 07:25:11.875] Read 0 show records
[INFO] [08-25-23 07:25:11.875] Starting TitleCardMaker (v1.14.3)
[DEBUG] [08-25-23 07:25:11.959] Latest remote version is v1.14.3
[INFO] [08-25-23 07:25:11.962] Read preference file "D:\SourceCode\TitleCardMaker\preferences.yml"
[DEBUG] [08-25-23 07:25:11.988] Using "magick " ImageMagick command prefix
[DEBUG] [08-25-23 07:25:11.992] Reset PersistentDatabase[remote_assets.json]
[DEBUG] [08-25-23 07:25:11.992] Not verifying SSL connections for Jellyfin
[INFO] [08-25-23 07:25:12.024] Starting to sync to series YAML files..
[INFO] [08-25-23 07:25:12.101] Synced D:\SourceCode\TitleCardMaker\config\sync_series.yml from Jellyfin
[INFO] [08-25-23 07:25:12.103] Reading series YAML file "D:\SourceCode\TitleCardMaker\config\sync_series.yml"..
[ERROR] [08-25-23 07:25:14.439] Uncaught Exception while processing "Jujutsu Kaisen (2020)"
[DEBUG] [08-25-23 07:25:14.440] string indices must be integers, not 'str'
[TRACEBACK] Traceback (most recent call last):
  File "D:\SourceCode\TitleCardMaker\modules\Manager.py", line 418, in __run_serially
    self.__run(serial=True)
  File "D:\SourceCode\TitleCardMaker\modules\Manager.py", line 385, in __run
    self.add_new_episodes()
  File "D:\SourceCode\TitleCardMaker\modules\Manager.py", line 35, in inner
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\SourceCode\TitleCardMaker\modules\Manager.py", line 235, in add_new_episodes
    show.add_new_episodes()
  File "D:\SourceCode\TitleCardMaker\modules\Show.py", line 535, in add_new_episodes
    all_episodes = interface.get_all_episodes(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\SourceCode\TitleCardMaker\modules\JellyfinInterface.py", line 372, in get_all_episodes
    for episode in response['Items']:
                   ~~~~~~~~^^^^^^^^^
TypeError: string indices must be integers, not 'str'
[INFO] [08-25-23 07:25:14.443] Reading series YAML file "D:\SourceCode\TitleCardMaker\config\sync_series.yml"..
[INFO] [08-25-23 07:25:14.690] Wrote missing assets to "D:\SourceCode\TitleCardMaker\missing.yml"

**Comment by CollinHeist** [comment]
@JPsychotic That error will only happen if the Jellyfin API is not returning valid Episode data for that Series. I'll add something to catch the error, but the issue most likely is with the Jellyfin API.

**Comment by JPsychotic** [comment]
Okay thank you. Just tell me when I can test again :)

**Comment by CollinHeist** [comment]
@JPsychotic You can switch to and pull the develop branch, that error should just be logged now

**Comment by JPsychotic** [comment]
So I did test and this is what it logs:

[ERROR] [08-25-23 20:47:36.368] Jellyfin returned bad Episode data for Jujutsu Kaisen (2020)
[DEBUG] [08-25-23 20:47:36.370] response='Series not found'
[INFO] [08-25-23 20:47:36.370] jellyfin has no episodes for "Jujutsu Kaisen (2020)"

I even specified the `jellyfin_id` inside the series.yml file. I don't get how it doesn't find the series when creating cards but when its syncing it creates an entry for it...

**Comment by JPsychotic** [comment]
So I did a little bit of testing with the jellyfin API and when I call the `[server_url]/Shows/[Jujutsu Kaisen Id]/Episodes` via Postman I get a normal looking result with all the episodes and their data... :(

**Comment by CollinHeist** [comment]
@JPsychotic That is weird. When you tested with the API I assume you used the same Jellyfin ID you have specified there in the YAML? So the endpoint was:

```
https://jellyfin.local/Shows/15ecda55a0eb2b1bcb1e9d1b4b29a4ac/Episodes
```

**Comment by JPsychotic** [comment]
> @JPsychotic That is weird. When you tested with the API I assume you used the same Jellyfin ID you have specified there in the YAML? So the endpoint was:
> 
> ```
> https://jellyfin.local/Shows/15ecda55a0eb2b1bcb1e9d1b4b29a4ac/Episodes
> ```

Yes, that is correct!

**Comment by CollinHeist** [comment]
Well that's even more confusing, all TCM does is make a request to that endpoint (with some arguments). So I am not sure why Jellyfin would return different results each time. You can try changing your Episode data source to Sonarr / TMDb

**Comment by JPsychotic** [comment]
After a lot of debugging with `log.debug(...)` because I don't know python I got it fixed. Problem was that I deleted the show from jellyfin and readded it because the naming scheme of the season folders and episodes made problems when jellyfin fetched the metadata. Because of missing metadata TCM often cannot create cards (missing index e.g.).

TCM has a series_infos.json where it stores series information. This file had the old `jellyfin_id` of the series. After deleting this file everything works again. :)
